lib/internal/errors.js

```diff
@@ -737,6 +737,11 @@ E('ERR_NO_ICU', '%s is not supported on Node.js compiled without ICU');
 E('ERR_NO_LONGER_SUPPORTED', '%s is no longer supported');
 E('ERR_OUT_OF_RANGE', outOfRange);
 E('ERR_PARSE_HISTORY_DATA', 'Could not parse history data in %s');
+E('ERR_PATH_IS_DIRECTORY', (path) => {
+  if (require('internal/util/types').isUint8Array(path))
```

Comment:
Since this error will only be thrown in case it is either a `string` or `Uint8Array` it would be cheaper to test for the string case.